### PR TITLE
PR: Convert command line options from optparse to argparse

### LINKED ISSUE #5639: Use argparse instead of optparse
optparse is deprecated see: https://docs.python.org/2.7/library/optparse.html

https://github.com/spyder-ide/spyder/blob/master/spyder/app/cli_options.py

### PR DESCRIPTION
**Pull request by csabella**
Fixes #5639.

I tried to minimize the changes by keeping the return value from `cli_options.py` the same as it was before, even though `parse_args` returns one value instead of two.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@rlaverde, please review this one.

**Comment by ccordoba12** [comment]
Thanks @csabella for your contribution!

Could add a simple test for this? I mean, like testing that running in a terminal that `spyder --help` returns part of a given string?

**Comment by csabella** [comment]
@ccordoba12   I added some tests for `get_option`.  I'm not sure if this is what you were looking for.  There is already a `test_mainwindow` which calls `get_option`.

File: spyder/app/tests/test_cli_options.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    options, args = getopt('--window-title MyWindow'.split())
+    assert options.window_title == 'MyWindow'
+
+    options, args = getopt('-p myproject test_file.py another_file.py'.split())
+    assert options.open_project == 'myproject'
+    assert options.files == ['test_file.py', 'another_file.py']
+    assert args == ['test_file.py', 'another_file.py']
+
+    with pytest.raises(SystemExit):
+        options, args = getopt(['--version'])
```
Comment (rlaverde) [inline spyder/app/tests/test_cli_options.py]: While reviewing this PR, I noticed that spyder doesn't have a `--version` option 

Maybe we could added it (in other PR), @ccordoba12 ?

Comment (ccordoba12) [reply spyder/app/tests/test_cli_options.py]: Yep, sure. Please go for it, when you have some free time ;-)

**Pull request merged**